src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -15,30 +15,31 @@ internal class SslConnectionInfo
         public readonly int KeyExchangeAlg;
         public readonly int KeyExchKeySize = 0;
 
-        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher)
+        internal SslConnectionInfo(Interop.libssl.SSL_CIPHER cipher, string protocol)
         {
-            Protocol = (int) MapProtocol(cipher.algorithm_ssl);
+            Protocol = (int) MapProtocolVersion(protocol);
             DataCipherAlg = (int) MapCipherAlgorithmType(cipher.algorithm_enc);
             KeyExchangeAlg = (int) MapExchangeAlgorithmType(cipher.algorithm_mkey);
             DataHashAlg = (int) MapHashAlgorithmType(cipher.algorithm_mac);
             // TODO (Issue #3362) map key sizes
         }
 
-        private static SslProtocols MapProtocol(long algorithm)
+        private SslProtocols MapProtocolVersion(string protocolVersion)
         {
-            switch (algorithm)
+            switch (protocolVersion)
             {
-                case 1:
+                case "SSLv2":
```

Comment:
Note: in Windows SSLv2 is prohibited OS-wide (although via Registry it can be brought back AFAIK).
The code looks good (we should support SSLv2 in case someone really needs to use it (insecurely) but, for security purposes, I'd like to understand if SSLv2 is, by default, disabled in OpenSSL.